### DISCUSSION TITLE
Открытие расчетная циклов по условию (тип поля ярлык)

### DISCUSSION THREAD

**Discussion by Re-Requeste** [totumonline/totum-ru-issues-and-discussions#552] [Как это сделать?] [answered]
Добрый день.

Есть таблица циклов ALLJ, в ней присутствуют две расч. таблицы цикл. tableA и tableB.
В циклах ALLJ, вытаскиваю поля с помощью **"тип поля: ярлык"** с таблицы tableA.

Задача состоит в том, что для ролей доступны разные расч. табл. циклов, т.е. для роля **AA доступна таблица tableA** для роля **BB доступна таблица tableB**. При нажатие на любое поле, открывается tableA т.к. ярлык подтягивает оттуда данные. Возможно ли, чтобы для роля BB открывался таблица tableB по умолчанию, использовав тип поля ярлык?

Пытался решить проблему несколькими способами:
а) Выбрать **тип поля "Ярлык"** (информацию брать с таблицы tableA) _[как и было изначально]_ и в код действе поставить условия if по роли, и открывать уже ту таблицу которая подходит для роли
б) Вытаскивать файлы напрямую из таблицы по условиям роли через select/sectList
в) В скрытом поле хедера поставить условия, где по роли будет приписана именно нужная таблица, и открывать таблицу используя пункт а или же заполнять данные используя пункт б

**Но, вопрос сильно ли будет нагружаться система, если я использую один из вариантов ?**
Есть еще одна догадка, можно создать ярлык в дереве, но там только можно настроить фильтр...

**Comment by erpic-ru** [comment]
Кажется, что поле Ярлык не влияет на выбор расчетной таблицы. Если для роли будет закрыта таблица АА, то пользователь в нее и не попадет, а попадет в следующую по сорту

**Reply by Re-Requeste** [reply]
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/230e6ac4-84a7-47ba-9022-2f56d7a7830c)

Я тоже так подумал, но таблица не доступна , в дереве появляется по сорту следующая _(нужная)_ таблица, если кликнуть она откроется, но автоматический она не открывается. Возможно в настройках где то нужно указать ?

**Reply by erpic-ru** [reply, marked as answer]
Тогда нужно поправить код кнопки перехода в цикл (button_to_cycle) с условием по роли

**Reply by Re-Requeste** [reply]
Это если пользователь нажмет на кнопку "открыть" то да, он провалиться именно в ту таблицу которая необходимо
Если же пользователь просто кликает на на любое поле, где тип стоит "Ярлык" он провалиться именно в ту таблицу с которой берется данные.

Можно конечно все вытащить через селекты, поставть блоки на поля ,  открывать только через кнопку, но : 
1. не будет ли нагрузка на сервер
2. и как то не красиво смотрится заболоченные поля, что чисто по кнопке необходимо нажимать

**Reply by Re-Requeste** [reply]
Сделал, спасибо. Не был в курсе что ярлык в циклах связан с кнопкой  button_to_cycle , получилось благодарю.

**Reply by ttm-alexeypolunin** [reply]
Да, используется код button_to_cycle, когда пользователь в таблице циклов кликает дважды на заблокированном поле.

Есть иногда задача, когда нужно отправить его в другое место при клике на определенное поле. Например у вас там статусы документов и что бы при клике на них провалится именно в таблицу Документы.

В этом случае в табл Циклов у поля включается код-действия при клике и там-же пишется кастомный код-действия с нужным linkToTable.
